utils/management_commands.py

```diff
@@ -0,0 +1,59 @@
+#
+# Freesound is (c) MUSIC TECHNOLOGY GROUP, UNIVERSITAT POMPEU FABRA
+#
+# Freesound is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Freesound is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     See AUTHORS file.
+#
+
+import json
+import logging
+import os
+import sys
+import time
+
+from django.core.management.base import BaseCommand
+
+
+commands_logger = logging.getLogger('commands')
+
+
+class LoggingBaseCommand(BaseCommand):
+
+    start_time = None
+    command_name = None
+
+    def find_command_name(self):
+        for count, arg in enumerate(sys.argv):
+            if arg == 'manage.py':
+                return sys.argv[count + 1]
+        return None
+
+    def log_start(self, data=None):
```

Comment:
none of the times you call log_start you pass in data, I wonder if it's necessary? Perhaps we could just set start_time and command_name in `__init__`?

Reply:
You're right, but I think in this way it is more generic. Maybe in some cases you do want to pass some data at the start of the command and some data at the end. It's just a couple of lines of code so I think it is fine keeping it. But if you have a strong feeling against it I'm fine removing.